File: output/sonarqube_analysis/codeium_solutions/python/results/q8-maximum-subarray.py
# https://leetcode.com/problems/maximum-subarray/
# Problem: Maximum Subarray

#Problem Description:
#Given an integer array nums, find the subarray with the largest sum, and return its sum.

# Constraints:
# 1 <= nums.length <= 10^5
# -10^4 <= nums[i] <= 10^4

#Code Structure:

class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        
        curr_sum = max_sum = nums[0]
        for num in nums[1:]:
            curr_sum = max(num, curr_sum + num)
            max_sum = max(max_sum, curr_sum)
        return max_sum
    

File: output/sonarqube_analysis/codeium_solutions/php/results/q8-maximum-subarray.php
<?php
/** https://leetcode.com/problems/maximum-subarray */
// Problem: Maximum Subarray

//Problem Description:
//Given an integer array nums, find the subarray with the largest sum, and return its sum.

// Constraints:
// 1 <= nums.length <= 10^5
// -10^4 <= nums[i] <= 10^4

//Code Structure:

class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function maxSubArray($nums) {
        $maxSoFar = $maxEndingHere = $nums[0];
        for ($i = 1; $i < count($nums); $i++) {
            $maxEndingHere = max($nums[$i], $maxEndingHere + $nums[$i]);
            $maxSoFar = max($maxSoFar, $maxEndingHere);
        }
        return $maxSoFar;
    }
}
    